plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php

```diff
@@ -408,12 +410,33 @@ protected function submitMetadata(KalturaCortexApiDistributionJobProviderData $a
 			$metadata["CoreField.Creation-Date:"] = $fieldValues[CortexApiDistributionField::MEDIA_CREATION_DATE] ? date('Y-m-d', $fieldValues[CortexApiDistributionField::MEDIA_CREATION_DATE]) : '';
 			$metadata["MAY.Legacy-Keywords:"] = $fieldValues[CortexApiDistributionField::MEDIA_KEYWORDS] ? str_replace(',', ', ', $fieldValues[CortexApiDistributionField::MEDIA_KEYWORDS]) : '';
 			$metadata["MAY.Job-Number:"] = $metadataFields[self::CORTEX_KALTURA_METADATA_FIELD_JOB_NUMBER];
-			$result = $this->requestCortex($metadata, self::CORTEX_API_SEND_METADATA);
+			$result = $this->requestCortex($metadata, $this->getBaseApiByMediaType($apiDistributionJobProviderData, false));
 			$this->setRecordId($result->RecordID ?? '');
 		}
 		catch(Exception $e)
 		{}
 	}
+	/**
+	 * @param KalturaCortexApiDistributionJobProviderData $apiDistributionJobProviderData
+	 * @param bool                                        $readApi
+	 * @return string
+	 */
+	protected function getBaseApiByMediaType(KalturaCortexApiDistributionJobProviderData $apiDistributionJobProviderData, bool $readApi)
+	{
+		if($apiDistributionJobProviderData->mediaType == KalturaMediaType::AUDIO)
```

Comment:
@TalDubovKaltura  this if complex looks inelegant to me - perhaps it would be better to keep a map such as 
`public static $apiEndpointMap = [
         KalturaMediaType::AUDIO => [
                               'get' => 'API/v2.2/DataTable/Documents.Video.General-Library-Audio:Read',
                               'update' => 'API/v2.2/DataTable/Documents.Video.General-Library-Audio:Update'
        ......
];`

Which would help help simplify this function into a one liner like 
`return self::$apiEndpointMap[$apiDistributionJobProviderData->mediaType][$readApi ? 'get' : 'update']`

Reply:
@hilak I even made it simpler now- https://github.com/kaltura/server/pull/13719/commits/e66dde57794eac0a8db7bd312846b967c52d1919
This way, in the future if we use a different version of the API than we should change it in only one place